src/source.ts

```diff
@@ -32,13 +33,30 @@ export default class Source extends React.Component<Props, void> {
 
     if (!map.getSource(this.id) && (geoJsonSource || tileJsonSource)) {
       map.addSource(this.id, (geoJsonSource || tileJsonSource) as any);
+      map.on('load', this.onData);
```

Comment:
I mean, there could be some issue with mapbox events (from their side), but the data is never available until map.on('load').

After that, you'll find source._data contains the actual data and is therefore not empty.

There are some related issues on mapbox (some of which are claimed as closed):

* https://github.com/mapbox/mapbox-gl-js/issues/3970
* https://github.com/mapbox/mapbox-gl-js/issues/3958

I tried to wait for `source.load` but it is like it never happens.